setup.py

```diff
@@ -318,555 +340,566 @@ def find_directories(self, location, use_pkgconfig=False):
                     # then, the directory will be returned as '\stuff' (!!)
                     # F. Alted 2006-02-16
                     if idx == 0:
-                        directories[idx] = os.path.dirname(path[:path.rfind(name)])
+                        # TODO: tweak this to make the above comment obsolete
+                        directories[idx] = Path(
+                            str(path)[: str(path).rfind(name)]
+                        )
```

Comment:
+1
Looks safe

Reply:
This is really a string hack, but it should be translatable to `Path` operations as well. I think I have understood the comments, but it should be clear without them as well.